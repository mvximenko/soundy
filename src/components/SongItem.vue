<template>
  <li
    class="
      flex
      justify-between
      items-center
      p-3
      pl-6
      cursor-pointer
      transition
      duration-300
      hover:bg-gray-50
      example
      snapshot
      song-id-abc
    "
  >
    <div>
      <router-link
        :to="{ name: 'Song', params: { id: song.docID } }"
        class="font-bold block text-gray-600"
        >{{ song.modified_name }}</router-link
      >
      <span class="text-gray-500 text-sm">{{ song.display_name }}</span>
    </div>

    <div class="text-gray-600 text-lg">
      <router-link
        custom
        :to="{ name: 'Song', params: { id: song.docID }, hash: '#comments' }"
        v-slot="{ navigate }"
      >
        <span class="comments" @click="navigate">
          <i class="fa fa-comments text-gray-600"></i>
          {{ song.comment_count }}
        </span>
      </router-link>
    </div>
  </li>
</template>

<script>
export default {
  name: 'SongItem',
  props: ['song'],
};
</script>
