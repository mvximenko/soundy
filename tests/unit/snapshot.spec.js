import { shallowMount, RouterLinkStub } from '@vue/test-utils';
import SongItem from '@/components/SongItem.vue';

describe('Snapshots SongItem.vue', () => {
  test('renders correctly', () => {
    const song = {
      docID: 'abc',
      modified_name: 'test',
      display_name: 'test',
      comment_count: 5,
    };

    const wrapper = shallowMount(SongItem, {
      propsData: { song },
      global: {
        components: {
          'router-link': RouterLinkStub,
        },
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
