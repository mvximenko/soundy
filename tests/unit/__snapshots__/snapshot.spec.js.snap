// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots SongItem.vue renders correctly 1`] = `
<li
  class="flex justify-between items-center p-3 pl-6 cursor-pointer transition duration-300 hover:bg-gray-50 example snapshot song-id-abc"
>
  <div>
    <router-link-stub
      class="font-bold block text-gray-600"
      to="[object Object]"
    />
    <span
      class="text-gray-500 text-sm"
    >
      test
    </span>
  </div>
  <div
    class="text-gray-600 text-lg"
  >
    <router-link-stub
      custom=""
      to="[object Object]"
    />
  </div>
</li>
`;
